src/api/status/public/js/build-log/build-header.js

```diff
@@ -0,0 +1,47 @@
+const autodeploymentUrl = (path) => `//dev.api.telescope.cdot.systems:4000/${path}`;
+
+const resMsg = document.getElementById('response-msg');
+const buildHeaderInfo = document.getElementById('build-header-info');
+const buildSender = document.getElementById('build-sender');
+const buildSenderName = document.getElementById('build-sender-name');
+const buildSenderImg = document.getElementById('build-sender-img');
+const buildGitSHA = document.getElementById('build-git-sha');
+const buildResult = document.getElementById('build-result');
+const buildStarted = document.getElementById('build-started');
+const buildDuration = document.getElementById('build-duration');
+
+function renderBuildInfo({ githubData, startedDate, stopDate, code }) {
+  resMsg.toggleAttribute('hidden');
+  buildHeaderInfo.toggleAttribute('hidden');
+  buildSender.setAttribute('href', githubData.sender.html_url);
+  buildSenderName.innerText = githubData.sender.login;
+  buildSenderImg.src = githubData.sender.avatar_url;
+  buildGitSHA.setAttribute('href', githubData.compare);
+  buildGitSHA.innerText = githubData.after.substring(0, 7);
+  buildResult.innerText = code === 0 ? 'Good' : 'Error';
+  buildStarted.innerText = new Date(startedDate).toUTCString();
+
+  const duration = new Date(stopDate).getTime() - new Date(startedDate).getTime();
+  const minutes = Math.floor(duration / 60000);
+  const seconds = ((duration % 60000) / 1000).toFixed(0);
+  buildDuration.innerText = `${minutes}m ${seconds}s`;
+}
+
+export default async function buildHeader() {
+  try {
+    const res = await fetch(autodeploymentUrl('status'));
+    if (!res.ok) {
+      const icon = document.createElement('i');
+      icon.className = 'fas fa-server px-2';
+      resMsg.innerHTML = '';
+      resMsg.append(icon);
+      resMsg.innerHTML += 'Unable to get build info.';
+
+      throw new Error('Unable to get build info');
+    }
+    const { previous } = await res.json();
+    renderBuildInfo(previous);
+  } catch (err) {
+    console.error(err);
+  }
+}
```

Comment:
We don't need to double fetch from "/status". The `checkBuildStatus` is already called every 5 seconds. You can just pass down the `githubData` and call `renderBuildInfo` inside `checkForBuild` when there is an ongoing build.

Reply:
That'll also keep the build-log and build-header in sync.

Reply:
Previously, I tried to follow #2486 to test if the data was fetch successfully, but now I'm getting this error:

<img width="902" alt="Screen Shot 2021-12-08 at 2 00 40 PM" src="https://user-images.githubusercontent.com/59778930/145267874-5f7bcb3a-0f75-4ef8-80d9-9954d99aec28.png">

Reply:
Content Security Error, you are missing this change.  But this is only for testing, do not include it in the commit.
```diff
-         connectSrc: [
-          "'self'",
-          '*.fontawesome.com',
-          `${host.replace(/(^\w+:|^)\/\//, '')}:4000`,
-          '*.github.com',
-        ],
+ connectSrc: ["'self'", "*"]

```